boardswarm/src/hifive_p550_mcu.rs

```diff
@@ -0,0 +1,371 @@
+        self.registrations.remove(device);
+    }
+}
+
+async fn setup_hifive_p550_mcu(prereg: PreRegistration, node: PathBuf, properties: Properties) {
+    info!(
+        "Setting up Hifive P550 MCU serial connection for {}",
+        node.display()
+    );
+    let port = match tokio_serial::new(node.to_string_lossy(), 115200).open_native_async() {
+        Ok(port) => port,
+        Err(e) => {
+            warn!("Failed to open serial port: {e}");
+            return;
+        }
+    };
+
+    let (sender, exec) = tokio::sync::mpsc::channel(16);
+    tokio::spawn(process(port, exec));
+
+    prereg.register_batch(|r| {
+        for command in McuCommand::iter() {
+            let mut properties = properties.clone();
+            properties.insert(registry::NAME, command.to_string());
+            let actuator = HifiveP550MCUActuator {
+                command,
+                sender: sender.clone(),
+            };
+            r.register_actuator(properties, actuator);
+        }
+    });
+}
+
+type ChannelMessage = (
+    McuCommand,
+    Box<dyn erased_serde::Deserializer<'static> + Send>,
+    tokio::sync::oneshot::Sender<Result<(), McuError>>,
+);
+
+async fn mcu_send(write: &mut WriteHalf<SerialStream>, request: &str) -> Result<(), McuError> {
+    debug!("Sending: {:?}", request);
+    write
+        .write_all(request.as_bytes())
+        .await
+        .map_err(|e| e.into())
+}
+
+async fn mcu_wait(
+    reader: &mut BufReader<ReadHalf<SerialStream>>,
+    response: &Regex,
+    timeout_millis: u64,
+) -> Result<String, McuError> {
+    let mut lines = reader.lines();
+    debug!("Waiting: {:?}", response);
+    loop {
+        match timeout(Duration::from_millis(timeout_millis), lines.next_line()).await {
+            Err(e) => {
+                return Err(e.into());
+            }
+            Ok(line) => match line {
+                Ok(Some(ret)) => {
+                    debug!("Response: {:?}", &ret);
+                    if response.is_match(&ret) {
+                        return Ok(ret);
+                    }
+                }
+                Ok(None) => {
+                    return Err(McuError::UnexpectedEof());
+                }
+                Err(e) => {
+                    return Err(e.into());
+                }
+            },
+        }
+    }
+}
+
+async fn process_som_power(
+    reader: &mut BufReader<ReadHalf<SerialStream>>,
+    write: &mut WriteHalf<SerialStream>,
+    parameters: SomPowerParameters,
+) -> Result<(), McuError> {
+    let (value, message) = parameters.mode.to_status();
+
+    {
+        let request = "sompower-g\n";
+        mcu_send(write, request).await?;
+    }
+
+    {
+        let response = Regex::new(r"^Som Power Status: (?<mode>.*)$").unwrap();
+        let ret = mcu_wait(reader, &response, MCU_RESPONSE_TIMEOUT_MILLIS).await?;
+        let caps = response.captures(&ret).unwrap();
+        if caps["mode"].to_lowercase() == parameters.mode.to_string() {
```

Comment:
```suggestion
       if caps["mode"].to_lowercase() == parameters.mode.as_ref() {
```
No need to allocate a string for the mode ; it's essentially static -- Fwiw this is where the AsRefStr derive comes into play